src/widgets/appointments/appointments-overview.component.tsx

```diff
@@ -34,6 +37,24 @@ const AppointmentsOverview: React.FC<AppointmentOverviewProps> = () => {
   const startDate = dayjs().format();
   const displayText = t("appointments", "appointments");
   const headerTitle = t("appointments", "Appointments");
+  const [pageNumber, setPageNumber] = React.useState(1);
+  const [pageSize, setPageSize] = React.useState(5);
+  const [currentPage, setCurrentPage] = React.useState([]);
```

Comment:
Same as above. I'd rather use `useMemo`. If you do that you could also easily introduce a hook like `usePaginate`.